File: RedSheet.DbEntities/Models/vCommunicationPlanRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vCommunicationPlanRecords", Schema = "dbo")]
	[Serializable]
	public partial class vCommunicationPlanRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CommunicationPlanId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string MediaMeans { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Message { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectPreparationId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string To { get; set; } 

	public vCommunicationPlanRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-communication-plan-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vCommunicationPlanRecord {
    constructor(vCommunicationPlanRecord?: vCommunicationPlanRecord )  {
        let properties = [ "communicationPlanId", "mediaMeans", "message", "projectPreparationId", "to",];
        for (let property of properties)
            if (vCommunicationPlanRecord && vCommunicationPlanRecord[property])
                this[property] = vCommunicationPlanRecord[property];
    }
 
	communicationPlanId : number =   0 ;
 
    @required()
    @maxLength(1000)
	mediaMeans : string =   undefined;
 
    @required()
    @maxLength(1000)
	message : string =   undefined;
 
    @range(1,2147483647)
	projectPreparationId : number =   undefined;
 
    @required()
    @maxLength(1000)
	to : string =   undefined;


}
